hash_cache.go

```diff
@@ -109,31 +107,28 @@ func (b *hashCacheS) TrieRootCached(val interface{}) ([32]byte, error) {
 // least recently added root info if the cache size has reached the max cache
 // size limit.
 func (b *hashCacheS) MerkleHashCached(byteSlice [][]byte) ([]byte, error) {
```

Comment:
Why not return `[32]byte`?

Reply:
Hash cache isn't handled by this PR, I think it'll require another PR to update and ensure ti works